webapp/channels/src/components/admin_console/access_control/policy_details/policy_details.tsx

```diff
@@ -0,0 +1,412 @@
+import TableEditor from 'components/admin_console/access_control/table_editor/table_editor';
+import BlockableLink from 'components/admin_console/blockable_link';
+import BooleanSetting from 'components/admin_console/boolean_setting';
+import Card from 'components/card/card';
+import TitleAndButtonCardHeader from 'components/card/title_and_button_card_header/title_and_button_card_header';
+import ChannelSelectorModal from 'components/channel_selector_modal';
+import SaveButton from 'components/save_button';
+import AdminHeader from 'components/widgets/admin_console/admin_header';
+import TextSetting from 'components/widgets/settings/text_setting';
+
+import {getHistory} from 'utils/browser_history';
+
+import './policy_details.scss';
+
+const DEFAULT_PAGE_SIZE = 10;
+
+interface PolicyActions {
+    fetchPolicy: (id: string) => Promise<ActionResult>;
+    createPolicy: (policy: AccessControlPolicy) => Promise<ActionResult>;
+    deletePolicy: (id: string) => Promise<ActionResult>;
+    searchChannels: (id: string, term: string, opts: ChannelSearchOpts) => Promise<ActionResult>;
+    setNavigationBlocked: (blocked: boolean) => void;
+    assignChannelsToAccessControlPolicy: (policyId: string, channelIds: string[]) => Promise<ActionResult>;
+    unassignChannelsFromAccessControlPolicy: (policyId: string, channelIds: string[]) => Promise<ActionResult>;
+    getAccessControlExpressionAutocomplete: () => Promise<ActionResult>;
+}
+
+export interface PolicyDetailsProps {
+    policyId?: string;
+    actions: PolicyActions;
+    autocompleteResult: AccessControlExpressionAutocomplete;
+}
+
+interface ChannelChanges {
+    removed: Record<string, ChannelWithTeamData>;
+    added: Record<string, ChannelWithTeamData>;
+    removedCount: number;
+}
+
+const PolicyDetails: React.FC<PolicyDetailsProps> = ({policyId, actions}) => {
+    const [policyName, setPolicyName] = useState('');
+    const [expression, setExpression] = useState('');
+    const [autoSyncMembership, setAutoSyncMembership] = useState(false);
+    const [serverError, setServerError] = useState(false);
+    const [addChannelOpen, setAddChannelOpen] = useState(false);
+    const [editorMode, setEditorMode] = useState<'cel' | 'table'>('cel');
+    const [channelChanges, setChannelChanges] = useState<ChannelChanges>({
+        removed: {},
+        added: {},
+        removedCount: 0,
+    });
+    const [saveNeeded, setSaveNeeded] = useState(false);
+    const [channelsCount, setChannelsCount] = useState(0);
+    const [autocompleteResult, setAutocompleteResult] = useState<AccessControlExpressionAutocomplete>({entities: {}});
+
+    useEffect(() => {
+        loadPage();
+    }, [policyId]);
+
+    const loadPage = async () => {
+        if (policyId) {
+            const result = await actions.fetchPolicy(policyId);
+
+            // Set policy name and expression after fetching the policy
+            setPolicyName(result.data?.name || '');
```

Comment:
Can result.data be null here? If so, do we want to handle it here? I'm thinking it could be null when the API call fails.

Reply:
Updated it to use `then` callback